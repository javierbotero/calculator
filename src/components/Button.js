/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import PropTypes from 'prop-types';

export default function Button({
  name, myClass, myHandler, color, wide,
}) {
  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    <div
      className={`${myClass} ${color}`}
      onClick={() => myHandler(name)}
      style={{
        backgroundColor: color,
        flexBasis: wide === 'true' ? '50%' : '25%',
      }}
    >
      {name}
    </div>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  myHandler: PropTypes.func.isRequired,
  myClass: PropTypes.string.isRequired,
  color: PropTypes.string,
  wide: PropTypes.string,
};

Button.defaultProps = {
  color: '#f5933d',
  wide: 'false',
};
