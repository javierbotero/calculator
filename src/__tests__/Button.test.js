import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

describe('Testing Button', () => {
  const handler = jest.fn();
  it('Checks the handler function is called once', () => {
    render(<Button name="AC" myHandler={handler} myClass="class" color="orange" wide="false" />);
    fireEvent.click(screen.getByText('AC'));
    expect(handler.mock.calls.length).toBe(1);
  });

  it('Makes sure when prop wide has the string value "true" the inline style has the value "flex-basis: 50%"', () => {
    render(<Button name="0" myHandler={handler} myClass="class" color="orange" wide="true" />);
    const button = screen.getByText('0');
    expect(button).toHaveAttribute('style', expect.stringContaining('flex-basis: 50%'));
  });

  it('Checks that button has class "orange" when prop color is equal to "orange"', () => {
    render(<Button name="x" myHandler={handler} myClass="class" color="orange" wide="true" />);
    const button = screen.getByText('x');
    expect(button).toHaveClass('orange');
  });

  it('Make sure the render is the same with the snapshot', () => {
    const tree = renderer
      .create(<Button name="AC" myHandler={handler} myClass="class" color="orange" wide="false" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
