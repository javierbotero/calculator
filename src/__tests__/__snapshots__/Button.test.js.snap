// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button Make sure the render is the same with the snapshot 1`] = `
<div
  className="class orange"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "orange",
      "flexBasis": "25%",
    }
  }
>
  AC
</div>
`;
